cnd/src/swap_protocols/rfc003/alice/spawner.rs

```diff
@@ -1,114 +1,60 @@
 use crate::{
-    comit_client::Client,
-    network::DialInformation,
+    connector::Connector,
     swap_protocols::{
-        self,
         asset::Asset,
         dependencies::LedgerEventDependencies,
-        metadata_store::{self, Metadata, MetadataStore, Role},
         rfc003::{
+            self,
             alice::{self, State, SwapCommunication},
-            messages::ToRequest,
             state_machine,
-            state_store::{self, StateStore},
+            state_store::StateStore,
             CreateLedgerEvents, Ledger,
         },
-        SwapId,
     },
 };
 use futures::{sync::mpsc, Stream};
 use futures_core::{
     compat::Future01CompatExt,
     future::{FutureExt, TryFutureExt},
 };
-use http_api_problem::HttpApiProblem;
 use std::sync::Arc;
 
-#[derive(Debug)]
-pub enum Error {
-    Storage(state_store::Error),
-    Metadata(metadata_store::Error),
-}
-
-impl From<Error> for HttpApiProblem {
-    fn from(e: Error) -> Self {
-        use self::Error::*;
-        match e {
-            Storage(e) => e.into(),
-            Metadata(e) => e.into(),
-        }
-    }
-}
-
-pub trait AliceSpawner: Send + Sync + 'static {
-    fn spawn<AL: Ledger, BL: Ledger, AA: Asset, BA: Asset>(
+pub trait AliceSpawn: Send + Sync + 'static {
+    #[allow(clippy::type_complexity)]
+    fn alice_spawn<AL: Ledger, BL: Ledger, AA: Asset, BA: Asset>(
         &self,
-        id: SwapId,
-        bob_dial_info: DialInformation,
-        swap_request: Box<dyn ToRequest<AL, BL, AA, BA>>,
-    ) -> Result<(), Error>
-    where
+        swap_request: rfc003::Request<AL, BL, AA, BA>,
+        response: rfc003::Response<AL, BL>,
+    ) where
         LedgerEventDependencies: CreateLedgerEvents<AL, AA> + CreateLedgerEvents<BL, BA>;
 }
 
-impl<T: MetadataStore, S: StateStore, C: Client> AliceSpawner
-    for swap_protocols::alice::ProtocolDependencies<T, S, C>
+impl<S> AliceSpawn for Connector<S>
```

Comment:
:wrench: How about calling it `Spawn` and using `<Connector as alice::Spawn>::spawn(&deps, swap_request, response)` at the call site?

Reply:
:wrench: Also, maybe rename the file to the name of the trait.

Reply:
Both suggestions apply for Bob too.

Reply:
Nice!  I made a lot of design choices based on not knowing this syntax.  Cheers.  Will use.

Reply:
On closer examination this suggestion is not going to work.  We call `alice_spawn` directly on the Connector within another Connector method i.e., we don't currently have a trait bound to use this syntax.  Could add the trait bound but this series is actively trying to move _away_ from using generics on the Connector.

Will, however, rename the file to match the trait.  Also renaming the trait to `SpawnAlice`.

Reply:
New idea!  I think I am going to be able to use your syntax now ...

Reply:
The renaming makes it better too, don't worry too much about my suggestion.

Reply:
Raised an issue to further look into the code duplication between the spawners.  https://github.com/comit-network/comit-rs